### DISCUSSION TITLE
Using logging framework with log4j2

### DISCUSSION THREAD

**Discussion by nappps** [line/armeria#5396] [Q&A]
Hi, 
I'm using Armeria framework, and also know it contains logback for logging.
To use an application as a container in Cloudwatch, the monitoring service of WAS, I have to use the log in json format for parsing.
And in order to do that logback contlib library must be added and used, which has a security vulnerability and no new version has been released.

So I am removing the logback library from the gradle and adding the log4j2 library. I wonder if there is any plan to integrate log4j2 into the framework and release it.

**Comment by minwoox** [comment]
You can use any logging framework of your choice with Armeria. Armeria relies solely on SLF4J, which serves as a logging facade. More information can be found at: https://mvnrepository.com/artifact/com.linecorp.armeria/armeria/1.26.4

Please let us know if you encounter any issues while adapting log4j2.

**Reply by nappps** [reply]
Ok. What I was curious about is that when using armeria, log4j2 is excluded as an exception to gradle to use, and I wonder how not to do this.

```
configurations {
    configureEach {
        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-logging'
        exclude group: 'ch.qos.logback', module: 'logback-classic'
    }
}
```

As an additional question, I am supposed to use slf4j now, and I wonder if you plan to use slf4j2 as a default.

**Reply by minwoox** [reply]
> log4j2 is excluded as an exception to gradle to use, and I wonder how not to do this.

Where did you get this code from?

>  I wonder if you plan to use slf4j2 as a default.

There are no specific plans for that at the moment.
We intend to transition to SLF4J 2 when we set the minimum Java version to 11.
This change is necessary as the use of SLF4J 2 requires an update of Logback to version 1.4, which in turn requires JDK 11.

https://logback.qos.ch/dependencies.html

**Reply by ikhoon** [reply]
> I'm using Armeria framework, and also know it contains logback for logging.

To clarify the dependency, Armeria modules do not depend on Logback by default. `armeria-logback` is an opt-in module for better supporting Logback.
